frontend/src/actions/ocpActions.js

```diff
@@ -263,6 +268,39 @@ export const setTableColumns = (key, isAdding) => (dispatch, getState) => {
 export const tableReCalcValues = () => (dispatch, getState) => {
   const { page, perPage } = getState().ocp;
 
-  dispatch(getOCPSummary());
   dispatch(setOCPPageOptions(page, perPage));
 };
+
+export const buildFilterData = () => async (dispatch, getState) => {
+  try {
+    const { tableFilters, categoryFilterValue } = getState().ocp;
+
+    const params = dispatch(getRequestParams("ocp"));
+
+    const response = await API.get("/api/v1/ocp/filters", { params });
+    if (response.status === 200 && response?.data?.filterData?.length > 0) {
+      let data = cloneDeep(response.data.filterData);
+      for (let i = 0; i < tableFilters.length; i++) {
+        for (let j = 0; j < data.length; j++) {
+          if (tableFilters[i]["value"] === data[j]["key"]) {
+            data[j]["name"] = tableFilters[i]["name"];
+          }
+        }
+      }
```

Comment:
This is a deeply nested set of code. You don't need to do this, but it may make sense to add a brief comment explaining those lines.

This also appears to be repeated. Are there any plans to refactor this?

Reply:
Added the `name` field in the response and avoided these nested loops.